students/ptfbanks/Lesson04/pf_trigrams.py

```diff
@@ -0,0 +1,107 @@
+#!/sr/bin/env python
+#Lesson04 - pf_Trigrams
+#---- set-up--------------------
+import re
+import random
+import os
+
+#---- Uer Instructions/ Selections--------------------
+print('Thefolowing prepares a raw text essay on any given input text,'
+'based entirely on the content of that text. From the out put the user is'
+'required to add personal touch of formatting, punctuation and advisably '
+'a grammer check. - Be advised, no guarantee of passing grade is offered.')
+
+siz = int(input('Of how many words is the essay to be?___'))
+
+u_opt =input('Enter file name (sherlock.txt) on which to base the generated essay:----\n'
+            'or enter "none" if default (test) text is peferred.__')
+if u_opt == 'none':
+    s = ('One night--it was on the twentieth of March, 1888--I was '
+    'eturning from a journey to a patient (for I had now returned to '
+    'civil practice), when my way led me through Baker Street. As I '
+    'passed the well-remembered door, which must always be associated '
+    'in my mind with my wooing, and with the dark incidents of the '
+    'Study in Scarlet, I was seized with a keen desire to see Holmes '
+    'again, and to know how he was employing his extraordinary powers. '
+    'His rooms were brilliantly lit, and, even as I looked up, I saw '
+    'his tall, spare figure pass twice in a dark silhouette against '
+    'the blind. He was pacing the room swiftly, eagerly, with his head '
+    'sunk upon his chest and his hands clasped behind him. To me, who '
+    'knew his every mood and habit, his attitude and manner told their '
+    'own story. He was at work again. He had risen out of his '
+    'drug-created dreams and was hot upon the scent of some new '
+    'problem. I rang the bell and was shown up to the chamber which '
+    'had formerly been in part my own.')
+else:
+    alt_file = u_opt
+    with open(alt_file, 'r')as f:
+        s= f.read()
+        
+#-------Functional objects----------------------
+
+def gen_words(s):     #prepares input file
+#    s = s.lower()    # Convert to lowercases ('reads better with this turned off)
+    s = s.replace("\n", " ")
+    s = re.sub(r'[^a-zA-Z0-9\s\n\n]', ' ', s)  # Replace characters with spaces 
+    s = s.replace("\n", "")
+    words = [token for token in s.split(" ") if token != ""] # create token, remove empty
+#    print('check1:',s)   #Diagnostic Insert
+    return(words)
+
+def build_trigrams(words_in):     # builds trigrams
+    trigram_list = {}             # initiate dictionary
+    for i in range(len(words_in)-2):
+        pair = (words_in[i], words_in[i+1])
+        next = []
+        if pair in trigram_list:   # if key exists, append value
+            trigram_list[pair].append(words_in[i+2])
+        else:
+            next = [words_in[i+2]]# if new key add dictionary entry
+            trigram_list[pair]= next
```

Comment:
`next` is a python built-in; avoid over-writing built-ins by using a different name.